### PR TITLE
FR-1919: consent order not approved email notification

### PR DESCRIPTION
**Pull request by jakub-kozlowski**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/FR-1919

### Change description ###

- added respondent solicitor email notification for consent order not approved event

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/NotificationService.java [hunk 1]
```diff
@@ -56,9 +56,17 @@ public void sendConsentOrderMadeConfirmationEmail(CaseDetails caseDetails) {
         sendNotificationEmail(notificationRequestMapper.createNotificationRequestForAppSolicitor(caseDetails), uri);
     }
 
-    public void sendConsentOrderNotApprovedEmail(CaseDetails caseDetails) {
+    public void sendConsentOrderNotApprovedEmailToApplicantSolicitor(CaseDetails caseDetails) {
+        sendConsentOrderNotApprovedEmail(notificationRequestMapper.createNotificationRequestForAppSolicitor(caseDetails));
+    }
+
+    public void sendConsentOrderNotApprovedEmailToRespondentSolicitor(CaseDetails caseDetails) {
+        sendConsentOrderNotApprovedEmail(notificationRequestMapper.createNotificationRequestForRespSolicitor(caseDetails));
```
Comment (shaunganley) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/NotificationService.java:64]: I assume the name of the CCD fields are the same between consented and contested which allows us to use this method? or is there logic to determine where it should come from in CCD based on the case type?

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/NotificationsControllerTest.java [hunk 2]
```diff
@@ -736,6 +736,18 @@ public void whenShouldSendRespondentNotification_andCaseIsConsented_thenShouldNo
         verify(notificationService, never()).sendContestOrderNotApprovedEmailRespondent(any());
     }
 
+    @Test
+    public void givenConsentedCase_whenSendConsentOrderNotApproved_thenShouldTriggerConsentedRespondentEmail() {
+        when(featureToggleService.isRespondentSolicitorEmailNotificationEnabled()).thenReturn(true);
```
Comment (shaunganley) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/NotificationsControllerTest.java:741]: you'll need tests for the other logic flows here

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/NotificationService.java (hunk 1 above)
Comment (jakub-kozlowski) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/NotificationService.java:64]: It is calling a "consented" method, so need for other logic I think.

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/NotificationsControllerTest.java (hunk 2 above)
Comment (jakub-kozlowski) [reply src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/NotificationsControllerTest.java:743]: The other flow is tested in `givenContestedCase_whenShouldSendRespondentNotification_thenShouldTriggerRespondentEmail` - added a `verify` there to test the method is not getting called in contested case.

**Review by ocarleton — APPROVED** [review]

**Pull request merged**